doc/classes/Object.xml

```diff
@@ -212,21 +212,28 @@
 				Combined with [method _get] and [method _get_property_list], this method allows defining custom properties, which is particularly useful for editor plugins. Note that a property [i]must[/i] be present in [method get_property_list], otherwise this method will not be called.
 				[codeblocks]
 				[gdscript]
+				var internal_data = {}
+
 				func _set(property, value):
-				    if (property == "fake_property"):
+				    if property == "fake_property":
 				        print("Setting my property to ", value)
+				        internal_data["fake_property"] = value
+				        return true
 
 				func _get_property_list():
 				    return [
 				        { "name": "fake_property", "type": TYPE_INT }
 				    ]
 				[/gdscript]
 				[csharp]
+				Godot.Collections.Dictionary InternalData = new Godot.Collections.Dictionary();
```

Comment:
**TL;DR:** Either of the two suggestions I make below should work, and we can decide on guidelines for writing C# documentation later.

---

If it's internal data it should probably be private[^1], and then the naming convention is _camelCase.

```suggestion
				private Godot.Collections.Dictionary _internalData = new Godot.Collections.Dictionary();
```

However, if it's internal data I'm assuming it doesn't communicate with Godot directly and then it can be `System.Collections.Generic.Dictionary<string, Variant>`[^2] instead.

```suggestion
				private System.Collections.Generic.Dictionary<string, Variant> _internalData = new System.Collections.Generic.Dictionary<string, Variant>();
```

We could also use targeted-type `new()` to avoid repeating the type name. I don't think we've done that before in the documentation though, and I would like the rest of the @godotengine/dotnet team to weigh in.

[^1]: `private` is the default accessibility when none is specified, but we are usually explicit.
[^2]: We recommend using the BCL collections when the data is not used with Godot's API because it can be more efficient. See https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_collections.html#choose-a-collection.

Reply:
I went with the Dictionary, because it's shorter.